packages/quantic/force-app/main/default/lwc/quanticResultList/__tests__/quanticResultList.test.js

```diff
@@ -0,0 +1,339 @@
+  fetchMoreResults: jest.fn(),
+};
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildResultList: functionsMocks.buildResultList,
+      buildSearchStatus: functionsMocks.buildSearchStatus,
+      buildResultsPerPage: functionsMocks.buildResultsPerPage,
+      buildResultTemplatesManager: functionsMocks.buildResultTemplatesManager,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticResultList && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticResultList) {
+      element.setInitializationError();
+    }
+  };
+}
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticResultList,
+  'c-quantic-result-list',
+  defaultOptions
+);
+
+describe('c-quantic-result-list', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    prepareHeadlessState();
+  });
+
+  afterEach(() => {
+    cleanup();
+    resultListState = initialResultListState;
+    searchStatusState = initialSearchStatusState;
+    resultsPerPageState = initialResultsPerPageState;
+    isInitialized = false;
+  });
+
+  describe('constroller initialization', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+    });
+
+    it('should initialize the result list, search status, results per page, and result templates manager controllers with the correct engine and options', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.buildResultList).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildResultList).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          options: {
+            fieldsToInclude: defaultFieldsToInclude.split(','),
+          },
+        }
+      );
+      expect(functionsMocks.resultsListSubscriber).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildSearchStatus).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildSearchStatus).toHaveBeenCalledWith(
+        exampleEngine
+      );
+      expect(functionsMocks.searchStatusStateSubscriber).toHaveBeenCalledTimes(
+        1
+      );
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildResultsPerPage).toHaveBeenCalledWith(
+        exampleEngine
+      );
+      expect(functionsMocks.resultsPerPageSubscriber).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildResultTemplatesManager).toHaveBeenCalledTimes(
+        1
+      );
+      expect(functionsMocks.buildResultTemplatesManager).toHaveBeenCalledWith(
+        exampleEngine
+      );
```

Comment:
nitpick: I would add space between each controller assertion to make it easier to read